File: algorithms/2_medium/S/01663_smallest_string_with_a_given_numeric_value/solution.c
char* getSmallestString(int n, int k) {
    char *restrict res = malloc(((size_t)n + 1)  * sizeof(*res));
    if (res == NULL) { return NULL; }

    res[n] = '\0';

    for (int i = 0; i < n; i += 1) {
        res[i] = 'a';
    }

    k -= n;
    for (int i = n - 1; i >= 0; i -= 1) {
        if (k <= 25) {
            res[i] += k;
            break;
        } else {
            res[i] += 25;
            k -= 25;
        }
    }

    return res;
}

File: algorithms/2_medium/S/01663_smallest_string_with_a_given_numeric_value/solution.rs
impl Solution {
    pub fn get_smallest_string(n: i32, mut k: i32) -> String {
        let mut res = vec![b'a'; n as usize];
        k -= n;
        for b in res.iter_mut().rev() {
            if k <= 25 {
                *b += (k as u8);
                break;
            } else {
                *b += 25;
                k -= 25;
            }
        }

        String::from_utf8(res).unwrap()
    }
}
